backend/src/main/resources/db/migration/V1.5__update_sequence_entries_view_status_field.sql

```diff
@@ -0,0 +1,37 @@
+drop view if exists sequence_entries_view;
+
+create view sequence_entries_view as
+select
+    se.*,
+    sepd.started_processing_at,
+    sepd.finished_processing_at,
+    sepd.processed_data as processed_data,
+    sepd.processed_data || em.joint_metadata as joint_metadata,
+    sepd.errors,
+    sepd.warnings,
+    case
+        when se.released_at is not null then 'APPROVED_FOR_RELEASE'
+        when se.is_revocation then 'PROCESSED'
+        when sepd.processing_status = 'IN_PROCESSING' then 'IN_PROCESSING'
+        when sepd.processing_status = 'PROCESSED' then 'PROCESSED'
+        else 'RECEIVED'
+    end as status,
+    CASE
+        WHEN sepd.processing_status = 'IN_PROCESSING' THEN NULL
+        WHEN sepd.errors IS NOT NULL AND jsonb_array_length(sepd.errors) > 0 THEN 'HAS_ERRORS'
+        WHEN sepd.warnings IS NOT NULL AND jsonb_array_length(sepd.warnings) > 0 THEN 'HAS_WARNINGS'
+        ELSE 'NO_ISSUES'
+    END AS processing_result
+from
+    sequence_entries se
+    left join sequence_entries_preprocessed_data sepd on
+        se.accession = sepd.accession
+        and se.version = sepd.version
+        and sepd.pipeline_version = (select version from current_processing_pipeline)
+    left join external_metadata_view em on
+        se.accession = em.accession
+        and se.version = em.version;
+
+UPDATE sequence_entries_preprocessed_data
+SET processing_status = 'PROCESSED'
+WHERE processing_status IN ('HAS_ERRORS', 'FINISHED');
```

Comment:
Indeed a great catch, @corneliusroemer, @anna-parker and @fhennig!

When we deploy a new version, Kubernetes will usually do a rolling update where it first starts the new backend and then shuts down the old one, right? (cc @theosanderson) If yes, this means that the old backend could still write "HAS_ERRORS" and "AWAITING_APPROVAL" into the database after the new backend executed the UPDATE query and thereby put the database into an invalid state.